File: src/main/java/com/lessmarkup/dataobjects/NodeUserData.java
package com.lessmarkup.dataobjects;

import com.lessmarkup.interfaces.data.AbstractDataObject;
import com.lessmarkup.interfaces.recordmodel.RequiredField;

public class NodeUserData extends AbstractDataObject {
    private long nodeId;
    private long userId;
    @RequiredField
    private String settings;

    public long getNodeId() {
        return nodeId;
    }

    public void setNodeId(long nodeId) {
        this.nodeId = nodeId;
    }

    public long getUserId() {
        return userId;
    }

    public void setUserId(long userId) {
        this.userId = userId;
    }

    public String getSettings() {
        return settings;
    }

    public void setSettings(String settings) {
        this.settings = settings;
    }
}

File: src/main/scala/com/lessmarkup/dataobjects/NodeUserData.scala
/*
 * This Source Code Form is subject to the terms of the Mozilla Public License, v. 2.0.
 * If a copy of the MPL was not distributed with this file, You can obtain one at
 * http://mozilla.org/MPL/2.0/.
 */

package com.lessmarkup.dataobjects

import com.lessmarkup.interfaces.annotations.RequiredField
import com.lessmarkup.interfaces.data.DataObject

class NodeUserData (
  id: Long,
  var nodeId: Long,
  var userId: Long,
  @RequiredField
  var settings: String
  ) extends DataObject(id)
